### PR TITLE
Allow specifying custom profiles

### PR DESCRIPTION
**Pull request by afnanenayet**
Allow for users to specify a specific profile. I have a ci-specific profile, so it would be useful to be able to set a different profile in the action config.

### CODE REVIEW

**Comment by taiki-e** [comment]
Thanks for the PR!

The `--profile` option affect the directory from which the built binaries are output, so you have to adjust `/release` below.

https://github.com/taiki-e/upload-rust-binary-action/blob/609508cff1a3f2297c188c806101d75e7196b17e/main.sh#L244

https://github.com/taiki-e/upload-rust-binary-action/blob/609508cff1a3f2297c188c806101d75e7196b17e/main.sh#L251-L252

https://github.com/taiki-e/upload-rust-binary-action/blob/609508cff1a3f2297c188c806101d75e7196b17e/main.sh#L263

See [this code](https://github.com/taiki-e/cargo-llvm-cov/blob/b72de10001e04ecad1530a32d221e7b828e8bfab/src/main.rs#L632-L639) in cargo-llvm-cov for the directory name used.

**Comment by afnanenayet** [comment]
Good catch, thanks

File: main.sh [hunk 1]
```diff
@@ -235,16 +241,16 @@ do_strip() {
 case "${INPUT_TARGET:-}" in
     '')
         build
-        target_dir="${target_dir}/release"
+        target_dir="${target_dir}/${INPUT_PROFILE}"
```
Comment (taiki-e) [inline main.sh:244]: Some profiles use different directories than the profile name, so you have to handle those.

- release, bench -> release
- dev, test -> debug
- (other) -> (profile name)

See also cargo-llvm-cov code linked in the above comment.

Comment (afnanenayet) [reply main.sh]: I updated the code so it'll rename the profile for the special cases

File: main.sh [hunk 2]
```diff
@@ -167,7 +167,26 @@ case "${OSTYPE}" in
...
+    input_profile="debug"
+fi
+
+declare -a build_options=()
+
+if [[ -n "${input_profile}" ]]; then
+    build_options+=("--profile" "${input_profile}")
+else
+    build_options+=("--release")
+fi
```
Comment (taiki-e) [inline main.sh:189]: `--profile` needs to use the profile name specified by the user, so the order of these two if-blocks needs to be swapped.

Comment (afnanenayet) [reply main.sh]: I split out two variables: `profile` and `profile_directory` to keep things a little clearer.

File: main.sh [hunk 3]
```diff
@@ -183,10 +206,10 @@ manifest_path="${INPUT_MANIFEST_PATH:-}"
 if [[ -n "${manifest_path}" ]]; then
     build_options+=("--manifest-path" "${manifest_path}")
     metadata=$(cargo metadata --format-version=1 --no-deps --manifest-path "${manifest_path}")
-    target_dir=$(jq <<<"${metadata}" -r '.target_directory')
+    target_dir=$(jq <<<"${metadata}" -r '.profile_directory')
 else
     metadata=$(cargo metadata --format-version=1 --no-deps)
-    target_dir=$(jq <<<"${metadata}" -r '.target_directory')
+    target_dir=$(jq <<<"${metadata}" -r '.profile_directory')
```
Comment (taiki-e) [inline main.sh:212]: Please revert them. profile_directory is not vaild field name of metadata.

**Review by taiki-e — APPROVED** [review]
Thanks!

**Comment by taiki-e** [comment]
Published in 1.14.0.

**Pull request merged**